pkg/providers/instance/instance.go

```diff
@@ -700,6 +702,38 @@ func (p *DefaultProvider) handleResponseErrors(ctx context.Context, instanceType
 
 		return fmt.Errorf("unable to allocate resources in the selected zone (%s). (will try a different zone to fulfill your request)", zone)
 	}
+	// TODO - should we be handling this error? Block specific VM Size in all zones?
+	// AllocationFailure means that VM allocation to the dedicated host has failed. But it can also mean "Allocation failed. We do not have sufficient capacity for the requested VM size in this region."
+	if sdkerrors.AllocationFailureOccurred(err) {
+		// TODO - this gives us duplicate keys that we try to mark in cache, can probably refactor it to avoid unnecessary work
```

Comment:
Left this as is for now, since I wasn't sure if we want to keep this code. Will remove the extra TODOs after initial review :)

Reply:
Fixed - now it only calls MarkUnavailable once for each combination of zone + instance type + capacity type